src/rtcan.c

```diff
@@ -209,6 +191,63 @@ rtcan_status_t rtcan_init(rtcan_handle_t* rtcan_h,
  */
 rtcan_status_t rtcan_start(rtcan_handle_t* rtcan_h)
 {
+    /* Since the subscriber ID list is known, we can configure the CAN ID Filter now */
+
+    uint32_t can_id_list[NUM_FILTERS * NUM_IDS_PER_FILTER] = {0};
+    int number_ids = 0;
+
+    /* Save and count all CAN IDs stored in hashmap */
+    for (uint32_t i = 0; i < RTCAN_HASHMAP_SIZE; i++)
```

Comment:
Loop iteration limit should be `sizeof(can_id_list) / sizeof(uint32_t)` or `NUM_FILTERS * NUM_IDS_PER_FILTER`. If you want to go for the second, I might make a new constant `TOTAL_FILTER_CAN_IDS` or something to use in both the `can_id_list_size` and the loop.

Reply:
Are you sure this should be `NUM_FILTERS * NUM_IDS_PER_FILTER`? This would be 28*4=112 which is bigger than size of hashmap. The `i` is used only for hashmap entries, not counting can_ids. Instead, I can put break statement in case 'number ids` will be bigger than `NUM_FILTERS * NUM_IDS_PER_FILTER`.

Reply:
My mistake, you're absolutely right. Honestly no idea what I was thinking sorry!